mautic/mautic
# Feature max daily sent emails

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | -
| New feature? | Y
| Related user documentation PR URL | -
| Related developer documentation PR URL | -
| Issues addressed (#s or URLs) | -
| BC breaks? | -
| Deprecations? | -

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

User can set "Max Sends per day" limit to Send Email Action/Event. Emails which will exceed daily limit will be waiting for next day to send. Any farther Actions/Decision/Conditions will be waiting (same as scheduled). For "0" value it will be sending emails without any limits.

#### Steps to test this PR:
1. Create few Leads (with same email adress).
2. Create Segment and attach Leads to it.
3. Create Campaign and add "Send Email" action to crated Segment in "Lunch campaign builder" with "Max Sends per day" different than 0.
4. Trigger Campaign multiple times. You should recive limited number of emails.
5. Open Lead details site (via Campaign details) and check informations about "queued" emails.
6. Change "date" field or remove entry from "campaign_event_daily_send_log" table in database (or wait until next day :) ) then trigger campaign again. You should recive rest of emails.

![1491309837815_original](https://cloud.githubusercontent.com/assets/3649580/24699199/8295d17a-19f3-11e7-8ee2-a399292e1fea.jpg)


![1491312504641_original](https://cloud.githubusercontent.com/assets/3649580/24699200/829699e8-19f3-11e7-900f-8d53c7b09d41.jpg)


![1491312480045_original](https://cloud.githubusercontent.com/assets/3649580/24699198/82958314-19f3-11e7-8aa9-fbc702292f2a.jpg)
